Orange/data/domain.py

```diff
@@ -148,7 +148,12 @@ def __init__(self, attributes, class_vars=None, metas=None, source=None):
                         raise TypeError(
                             "descriptors must be instances of Variable, "
                             "not '%s'" % type(var).__name__)
-
+      
+        names = [var.name for var in chain(attributes, class_vars, metas)]
+        if len(names) != len(set(names)):
```

Comment:
Maybe add a test for this exception? It does seem pointless, but it may prevent somebody to accidentaly remove or circumvent this check. And it doesn't cost a lot. And it increases the coverage.